sdk/azcore/policy_body_download.go

```diff
@@ -28,8 +28,9 @@ func newBodyDownloadPolicy() Policy {
 			resp.Body.Close()
 			if err != nil {
 				err = fmt.Errorf("body download policy: %w", err)
```

Comment:
If there's an error here, wouldnt we want to return immediately here?

Reply:
Fixed.